scipy/stats/_stats_py.py

```diff
@@ -2821,26 +2821,30 @@ def obrientransform(*samples):
     that the variances are different.
 
     """
-    TINY = np.sqrt(np.finfo(float).eps)
+    xp = array_namespace(*samples)
+    default_float = xp.asarray(1.).dtype
+    TINY = math.sqrt(xp.finfo(default_float).eps)
```

Comment:
Should this be the dtype of `sampe` instead?

Reply:
Yup, looks like it.